bee-network/bee-autopeering/src/server.rs

```diff
@@ -0,0 +1,290 @@
+    const NAME: &'static str = "IncomingPacketHandler";
+    const SHUTDOWN_PRIORITY: u8 = 2;
+
+    type ShutdownSignal = ShutdownRx;
+
+    async fn run(self, mut shutdown_rx: Self::ShutdownSignal) {
+        let IncomingPacketHandler {
+            incoming_socket,
+            incoming_senders,
+            bind_addr,
+        } = self;
+
+        let mut packet_bytes = [0; READ_BUFFER_SIZE];
+
+        let IncomingPacketSenders {
+            discovery_tx,
+            peering_tx,
+        } = incoming_senders;
+
+        'recv: loop {
+            tokio::select! {
+                _ = &mut shutdown_rx => {
+                    break;
+                }
+                r = incoming_socket.recv_from(&mut packet_bytes) => {
+                    match r {
+                        Ok((n, peer_addr)) => {
+                            if peer_addr == bind_addr {
+                                log::warn!("Received bytes from own bind address {}. Ignoring...", peer_addr);
+                                continue 'recv;
+                            }
+
+                            if n > MAX_PACKET_SIZE {
+                                log::warn!("Received too many bytes from {}. Ignoring packet.", peer_addr);
+                                continue 'recv;
+                            }
+
+                            log::trace!("Received {} bytes from {}.", n, peer_addr);
+
+                            let packet = Packet::from_protobuf(&packet_bytes[..n]).expect("error decoding incoming packet");
+                            // log::trace!("{} ---> public key: {}.", peer_addr, multiaddr::from_pubkey_to_base58(&packet.public_key()));
+
+                            // Restore the peer id.
+                            let peer_id = PeerId::from_public_key(packet.public_key());
+                            // log::trace!("{} ---> peer id: {}.", peer_addr, peer_id);
+
+                            // Verify the packet.
+                            let message = packet.message();
+                            let signature = packet.signature();
+                            if !packet.public_key().verify(&signature, message) {
+                                log::trace!("Received packet with invalid signature");
+                                continue 'recv;
+                            }
+
+                            let marshalled_bytes = packet.into_message();
+                            let (msg_type, msg_bytes) = unmarshal(&marshalled_bytes);
+
+                            let packet = IncomingPacket {
+                                msg_type,
+                                msg_bytes,
+                                peer_addr,
+                                peer_id,
+                            };
+
+                            // Depending on the message type, forward it to the appropriate manager.
+                            match msg_type as u32 {
+                                t if DISCOVERY_MSG_TYPE_RANGE.contains(&t) => {
```

Comment:
I guess I cannot use the "DISCOVERY_MSG_TYPE_RANGE" range expression, so I would have to use the pattern `a..b` directly, but then it would get even more lengthy due to the long const names, so I would like to keep it as it is.

From the Rust Reference:
"Note: The 2..=9 is a Range Pattern, not a Range Expression. Thus, only those types of ranges supported by range patterns can be used in match arms."